app/src/renderer/components/common/ManagedBody.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <div>
+    <modal-search
+      v-if="search && search.somethingToSearch"
+      :type="search.type"
+    />
+
+    <tm-data-connecting v-if="!loaded && !connected" />
+    <tm-data-loading v-else-if="!loaded && loading" />
+    <tm-data-error v-else-if="error" />
+    <slot
+      v-else-if="data.length === 0 && this.$slots['no-data']"
+      name="no-data"
+    />
+    <tm-data-empty v-else-if="data.length === 0" />
+    <data-empty-search v-else-if="filteredData.length === 0" />
+    <slot v-else name="data-body" />
+  </div>
+</template>
+
+<script>
+import { TmDataMsg, TmDataLoading, TmDataEmpty } from "@tendermint/ui"
+import DataEmptySearch from "common/TmDataEmptySearch"
+import TmDataError from "common/TmDataError"
+import ModalSearch from "common/TmModalSearch"
+import TmDataConnecting from "common/TmDataConnecting"
+export default {
+  name: `managed-body`,
```

Comment:
not sure i understand the naming choice here @sabau

Reply:
I wasn't sure neither honestly.
I was thinking of the goal of this component, basically it try to manage the statuses of data you pass to it, with a mandatory body slot, so I tried to express it somehow but the naming is not really clear.

Do you have some suggestion on how to make it change it?

It manages all the statuses in one place so I thought of manage something

Reply:
Oh wait, I just realized that this component will fit as sub component of tm-page, once the merge is complete, will simply define a new slot